PKHeX.Core.AutoMod/AutoMod/RegenTemplate.cs

```diff
@@ -28,7 +28,7 @@ public sealed class RegenTemplate : IBattleTemplate
         public Ball Ball { get; set; }
         public Shiny ShinyType { get; set; } = Core.Shiny.Random;
 
-        public RegenTemplate(IBattleTemplate set)
+        public RegenTemplate(IBattleTemplate set, int gen)
```

Comment:
One last modifcation that I'd suggest here is to default `int gen` to PKX.Generation (To always grab the latest generation if a generation argument isnt passed) :)